pkg/cvo/configuration/configuration.go

```diff
@@ -0,0 +1,111 @@
+package configuration
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	operatorclientset "github.com/openshift/client-go/operator/clientset/versioned"
+	cvoclientv1alpha1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1alpha1"
+	operatorexternalversions "github.com/openshift/client-go/operator/informers/externalversions"
+	operatorlistersv1alpha1 "github.com/openshift/client-go/operator/listers/operator/v1alpha1"
+)
+
+const ClusterVersionOperatorConfigurationName = "cluster"
+
+type ClusterVersionOperatorConfiguration struct {
+	queueKey string
+	// queue tracks checking for the CVO configuration.
+	//
+	// The type any is used to comply with the worker method of the cvo.Operator struct.
+	queue workqueue.TypedRateLimitingInterface[any]
+
+	client  cvoclientv1alpha1.ClusterVersionOperatorInterface
+	lister  operatorlistersv1alpha1.ClusterVersionOperatorLister
+	factory operatorexternalversions.SharedInformerFactory
+
+	started bool
+}
+
+func (config *ClusterVersionOperatorConfiguration) Queue() workqueue.TypedRateLimitingInterface[any] {
+	return config.queue
+}
+
+// clusterVersionOperatorEventHandler queues an update for the cluster version operator on any change to the given object.
+// Callers should use this with an informer.
+func (config *ClusterVersionOperatorConfiguration) clusterVersionOperatorEventHandler() cache.ResourceEventHandler {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc: func(_ interface{}) {
+			config.queue.Add(config.queueKey)
+		},
+		UpdateFunc: func(_, _ interface{}) {
+			config.queue.Add(config.queueKey)
+		},
+		DeleteFunc: func(_ interface{}) {},
```

Comment:
If there is an instance that configured us, and the instance got deleted, then we should `sync` and reset to some default. But maybe we can address this case later - can we add a todo?

Reply:
Could we make the logic to set default values when the resource is not found? To not make distinctions between `not found` vs. `got deleted`.  The motivation is to simplify the logic. 

The `DeleteFunc` would add the key to queue, the `sync` would set default values when `err == isNotFound`.

Reply:
Yeah, that sounds reasonable :+1:

Reply:
Updated the `DeleteFunc` and added a TODO.